pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -102,3 +162,23 @@ func (s *completed) Name() string {
 func (s *completed) CanTransitionTo(next state) bool {
 	return false
 }
+
+func (s *completed) Execute(msg dispatcher.DIDCommMsg) (state, error) {
+	switch msg.Type {
+	case connectionResponse:
+		if msg.Outbound {
+			return nil, fmt.Errorf("outbound responses are not allowed for state %s", s.Name())
+		}
+		// send ACK
+		return nil, nil
```

Comment:
I am seeing return nil, nil several times in the state machine. I also see a null struct. Is this supposed to be return *null{}, nil?

(I'm a bit confused.)

Reply:
nope - `null` != `nil`. Want me to rename `null`?

There are implicit TODOs in this file - like the line above that says "send ACK". If that results in an error then the return would change to `return nil, err`

Reply:
As per our discussion - I added a new special state `noOp` to aid clarity in states.go in cases where we previously had `return nil, nil`